pkg/apis/kubeone/validation/validation.go

```diff
@@ -269,7 +271,7 @@ func ValidateCloudProviderSpec(p kubeoneapi.CloudProviderSpec, networkConfig kub
 		allErrs = append(allErrs, field.Invalid(fldPath, "", "provider must be specified"))
 	}
 
-	if p.Vsphere == nil && len(p.CSIConfig) > 0 {
+	if providerSpec.Vsphere == nil && len(providerSpec.CSIConfig) > 0 {
```

Comment:
I think we should also reject CSIConfig if CSI drivers are disabled.

Reply:
done